docs/src/tutorial/minimal_working_compute.jl

```diff
@@ -0,0 +1,266 @@
+# (`..._HOST_VISIBLE_BIT`).
+memorytype_idx = 0
+
+# Let's create some data. We will work with 100 flimsy floats.
+data_items = 100
+mem_size = sizeof(Float32) * data_items
+
+# Allocate the memory of the correct type
+mem = unwrap(allocate_memory(device, mem_size, memorytype_idx))
+
+# Make a buffer that will be used to access the memory, and bind it to the
+# memory. (Memory allocations may be quite demanding, it is therefore often
+# better to allocate a single big chunk of memory, and create multiple buffers
+# that view it as smaller arrays.)
+buffer = unwrap(create_buffer(device, mem_size, BUFFER_USAGE_STORAGE_BUFFER_BIT, SHARING_MODE_EXCLUSIVE, [qfam_idx]))
+
+bind_buffer_memory(device, buffer, mem, 0)
+
+# ## Uploading the data to the device
+#
+# First, map the memory and get a pointer to it.
+memptr = map_memory(device, mem, 0, mem_size)
+
+# Here we force Julia to look at the mapped data as at the vector of
+# `Float32`s, so that we can access it easily.
+data = unsafe_wrap(Vector{Float32}, convert(Ptr{Float32}, unwrap(memptr)), data_items, own = false)
+
+# For now, let's just zero out all the data, and *flush* the changes to make
+# sure the device can see the updated data. This is the simplest way to move
+# array data to the device.
+data .= 0
+flush_mapped_memory_ranges(device, [MappedMemoryRange(mem, 0, mem_size)])
+
+# Eventually, you may need to allocate memory types that are not visible from
+# host, because these provide better capacity and speed on the discrete GPUs.
+# At that point, you may need to use the transfer queue and memory transfer
+# commands to get the data from host-visible to GPU-local memory, using e.g.
+# [`cmd_copy_buffer`](@ref).
+
+# ## Compiling the shader
+#
+# Now we need to make a shader program. We will use `glslangValidator` packaged
+# in a JLL to compile a GLSL program from a string into a spir-v bytecode,
+# which is later passed to the GPU drivers.
+
+shader_code = """
+#version 430
+
+layout (local_size_x_id = 0) in;
+layout (local_size_x = 1, local_size_y = 1, local_size_z = 1) in;
+
+layout(constant_id = 0) const uint blocksize = 1; // copy of local_size_x
+
+layout (push_constant) uniform Params
+{
+    float val;
+    uint n;
+} params;
+
+layout(std430, binding=0) buffer databuf
+{
+    float data[];
+};
+
+void
+main()
+{
+    uint i = gl_GlobalInvocationID.x;
+    if(i < params.n) data[i] = params.val * i;
+}
+"""
+
+# Push constants are small packs of variables that are used to quickly send
+# configuration data to the shader runs. Make sure that this structure
+# corresponds to what is declared in the shader.
+struct shader_push_consts
+    val::Float32
+    n::UInt32
+end
+
+# Specialization constants are similar to push constants, but less dynamic: You
+# can change them before compiling the shader for the pipeline, but not
+# dynamically. This may have performance benefits for "very static" values,
+# such as block sizes.
+struct shader_spec_consts
+    local_size_x::UInt32
+end
```

Comment:
We can rename them to `ShaderPushConsts` and `ShaderSpecConsts` to be in line with the standard naming convention for types in Julia.
(I read `spec_consts = [shader_spec_consts(const_local_size_x)]` and thought for a second that `shader_spec_consts` was a function and not a type)

Reply:
good point, added